test/test_optim.py

```diff
@@ -1548,7 +1548,11 @@ def getPublicAttrs(obj):
 
             # Make some state
             for _ in range(3):
-                optimizer.step(closure)
+                if torch._dynamo.is_compiling() and not optim_info.step_requires_closure:
+                    closure()
+                    optimizer.step()
+                else:
+                    optimizer.step(closure)
```

Comment:
This is more readable. Would be good to have this as the pattern everywhere.
```suggestion
                if optim_info.step_requires_closure:
                    optimizer.step(closure)
                else:
                    closure()
                    optimizer.step()
```